server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -62,41 +64,71 @@
 
     public static class Nested {
 
-        public static final Nested NO = new Nested(false, false, false);
+        public static final Nested NO = new Nested(false, new Explicit<>(false, false), new Explicit<>(false, false));
 
-        public static Nested newNested(boolean includeInParent, boolean includeInRoot) {
+        public static Nested newNested() {
+            return new Nested(true, new Explicit<>(false, false), new Explicit<>(false, false));
+        }
+
+        public static Nested newNested(Explicit<Boolean> includeInParent, Explicit<Boolean> includeInRoot) {
             return new Nested(true, includeInParent, includeInRoot);
         }
 
         private final boolean nested;
+        private Explicit<Boolean> includeInParent;
+        private Explicit<Boolean> includeInRoot;
 
-        private final boolean includeInParent;
-
-        private final boolean includeInRoot;
-
-        private Nested(boolean nested, boolean includeInParent, boolean includeInRoot) {
+        private Nested(boolean nested, Explicit<Boolean> includeInParent, Explicit<Boolean> includeInRoot) {
             this.nested = nested;
             this.includeInParent = includeInParent;
             this.includeInRoot = includeInRoot;
         }
 
+        public void merge(Nested mergeWith, MergeReason reason) {
+            if (isNested()) {
+                if (!mergeWith.isNested()) {
+                    throw new IllegalArgumentException("Cannot change object mapping from nested to non-nested.");
```

Comment:
Super minor nit, but most of our error messages are all lowercase and without ending punctuation and I think that's the precedent we want to stick with (applies to the errors below also)
```suggestion
                    throw new IllegalArgumentException("cannot change object mapping from nested to non-nested");
```

Reply:
Oops, I have been adding inconsistent error messages for quite some time now!